driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/execveat.bpf.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/variable_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(execveat_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct auxiliary_map *auxmap = auxmap__get();
+	if(!auxmap)
+	{
+		return 0;
+	}
+	auxmap__preload_event_header(auxmap, PPME_SYSCALL_EXECVEAT_E);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: dirfd (type: PT_FD) */
+	s32 dirfd = (s32)extract__syscall_argument(regs, 0);
+	if(dirfd == AT_FDCWD)
+	{
+		dirfd = PPM_AT_FDCWD;
+	}
+	auxmap__store_s64_param(auxmap, (s64)dirfd);
+
+	/* Parameter 2: pathname (type: PT_FSRELPATH) */
+	unsigned long pathname_pointer = extract__syscall_argument(regs, 1);
+	auxmap__store_charbuf_param(auxmap, pathname_pointer, USER);
+
+	/* Parameter 3: flags (type: PT_FLAGS32) */
+	unsigned long flags = extract__syscall_argument(regs, 4);
+	auxmap__store_u32_param(auxmap, execveat_flags_to_scap(flags));
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	auxmap__finalize_event_header(auxmap);
+
+	auxmap__submit_event(auxmap);
+	return 0;
+}
+
+/*=============================== ENTER EVENT ===========================*/
+
+/*=============================== EXIT EVENT ===========================*/
+
+SEC("tp_btf/sys_exit")
+int BPF_PROG(execveat_x,
+	     struct pt_regs *regs,
+	     long ret)
+{
+	struct auxiliary_map *auxmap = auxmap__get();
+	if(!auxmap)
+	{
+		return 0;
+	}
+	auxmap__preload_event_header(auxmap, PPME_SYSCALL_EXECVEAT_X);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	auxmap__store_s64_param(auxmap, ret);
+
+	struct task_struct *task = get_current_task();
+
+	/* This is a charbuf pointer array.
+	 * Every element of `argv` array is a pointer to a charbuf.
+	 * Here the first pointer points to `exe` param while all
+	 * the others point to the different args.
+	 *
+	 * Please note: this bpf program is called only if the `execveat` fails
```

Comment:
Yeah you can find more context in this PR https://github.com/falcosecurity/libs/pull/141

TL;DR;
`execveat` is a wrapper of `execve`, when the call succeeds the event returned is simply an `execve` exit event. Only in case of syscall failure, we can observe an `execveat` exit event...

Reply:
understood.. thanks for the background on it!